pkg/tici/tici_manager_client.go

```diff
@@ -565,3 +570,32 @@ func DropFullTextIndex(ctx context.Context, store kv.Storage, tableID int64, ind
 	}
 	return ticiManager.DropFullTextIndex(ctx, tableID, indexID)
 }
+
+// cloneAndNormalizeTableInfo deep-clones the given model.TableInfo via JSON
+// marshal/unmarshal and forces FieldType.Flen for longtext and json column types
+// through an int32 narrowing (int(int32(flen))). This ensures TiCI parses the
+// length as int32 and avoids 4GB default-value related errors.
+func cloneAndNormalizeTableInfo(tbl *model.TableInfo) (*model.TableInfo, error) {
```

Comment:
Since it is only used for JSON, I think we can wrap "Marshal" into this function and output JSON directly.